test/src/dftbp/api/mm/testers/test_mpisubgrids.f90

```diff
@@ -44,112 +44,131 @@ program test_mpisubgrids
 
   integer, parameter :: groupSize = 4
 
-  type(TDftbPlus) :: dftbp
-  type(TDftbPlusInput) :: input
+  call main_()
 
-  real(dp) :: merminEnergy, totalMerminEnergy, reducedMerminEnergy
-  real(dp) :: gradients(3, 3), totalGradients(3, 3), reducedGradients(3, 3)
-  integer :: devNull
-  integer :: providedThreading, iErr, myId, myGroup, nProc
-  integer :: iRepeat, nAtom
+contains
 
-  !type(mpi_comm) :: groupComm
-  integer :: groupComm
-  integer :: nGroup, myIdGroup, nProcGroup
 
-  logical :: doSi2
+  !! Main test routine
+  !!
+  !! All non-constant variables must be defined here to ensure that they are all explicitely
+  !! deallocated before the program finishes.
+  !!
+  subroutine main_()
 
-  open(newunit=devNull, file="/dev/null", action="write")
+    type(TDftbPlus), allocatable :: dftbp
+    type(TDftbPlusInput), allocatable :: input
 
-  call mpi_init_thread(requiredThreading, providedThreading, iErr)
-  call mpi_comm_rank(MPI_COMM_WORLD, myId, iErr)
-  call mpi_comm_size(MPI_COMM_WORLD, nProc, iErr)
+    real(dp) :: merminEnergy, totalMerminEnergy, reducedMerminEnergy
+    real(dp) :: gradients(3, 3), totalGradients(3, 3), reducedGradients(3, 3)
+    integer :: devNull
+    integer :: providedThreading, iErr, myId, myGroup, nProc
+    integer :: iRepeat, nAtom
 
-  if (modulo(nProc, groupSize) /= 0) then
-    error stop "Number of processors must be a multiple of the groupSize"
-  end if
+    !type(mpi_comm) :: groupComm
```

Comment:
Is there any reason to retain the commented line, also why is the integer preferred instead of the type?